src/main/java/com/aws/greengrass/device/exception/InvalidSessionException.java

```diff
@@ -0,0 +1,22 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.device.exception;
+
+public class InvalidSessionException extends Exception {
```

Comment:
Right, I thought about it but not sure why an invalid session should be tied to an AuthorizationException. But if we want to avoid refactoring moquette code for this, I'll make it subtype of `AuthorizationException`

Reply:
It is a type of authorization exception. At a high level, we cannot authorize the session. The reason why we can't authorize it is because there isn't any session. Adding it as a subclass, IMO is correct object modeling, and has the benefit of keeping the method's interface the same.